arangod/IResearch/IResearchFilterFactory.cpp

```diff
@@ -1437,7 +1569,8 @@ Result fromArrayComparison(irs::boolean_filter*& filter,
     }
     Result buildRes;
     aql::AstNodeType arrayExpansionNodeType;
-    std::tie(buildRes, arrayExpansionNodeType) =
+    size_t matchCount;
+    std::tie(buildRes, arrayExpansionNodeType, matchCount) =
```

Comment:
nit: could be made simpler by using `auto [buildRes, arrayExpansionNodeType, matchCount] = ...`
```suggestion
    std::tie(buildRes, arrayExpansionNodeType, matchCount) =
```